SwiftLeeds/Views/Components/FancyHeaderView.swift

```diff
@@ -0,0 +1,171 @@
+//
+//  HeaderView.swift
+//  SwiftLeeds
+//
+//  Created by Kannan Prasad on 05/07/2022.
+//
+
+import SwiftUI
+
+struct FancyHeaderView: View {
+    
+    internal init(title: String, foregroundImageURL: URL?, backgroundImageURL: URL?) {
+        self.title = title
+        self.foregroundImageURL = foregroundImageURL
+        self.backgroundImageURL = backgroundImageURL
+        self.foregroundImageName = nil
+        self.backgroundImageName = nil
+    }
+    
+    internal init(title: String, foregroundImageName: String?, backgroundImageName: String?) {
+        self.title = title
+        self.foregroundImageURL = nil
+        self.backgroundImageURL = nil
+        self.foregroundImageName = foregroundImageName
+        self.backgroundImageName = backgroundImageName
+    }
+    
+    let title: String
+    let foregroundImageURL: URL?
+    let backgroundImageURL: URL?
+    let foregroundImageName: String?
+    let backgroundImageName: String?
+    
+    private let foregroundImageWidth = 160.0
+    private let aspectRatio = 1.66
+    var shadowColor: Color {
+       Color.black.opacity(1/3)
+    }
+    
+    @State var foregroundGroupViewHeight: CGFloat = .zero
+    
+    var body: some View {
+        Rectangle()
+            .foregroundColor(.clear)
+            .edgesIgnoringSafeArea(.top)
+            .aspectRatio(aspectRatio, contentMode: .fill)
+            .background(backgroundImage
+                            .aspectRatio(contentMode: .fill))
+            .overlay(foregroundGroup,alignment: .center)
+            .padding(.bottom,foregroundGroupViewHeight/2)
+    }
+     
+    
+    @ViewBuilder
+    private var backgroundImage: some View {
+        if let backgroundImageURL = backgroundImageURL {
+           AsyncImage(url: backgroundImageURL) { phase in
+                switch phase {
+                case .empty:
+                    ProgressView()
+                case .success(let image):
+                    crateRectangleImage(for: image, aspectRatio: aspectRatio)
+                case .failure(_):
+                    crateRectangleImage(for: Image(Assets.Image.playhouseImage),
+                                           aspectRatio: aspectRatio)
+                @unknown default:
+                    ProgressView()
+                        .tint(.white)
+                        .opacity(0.5)
+                }
+            }
+        } else if let backgroundImageName =  backgroundImageName {
+            crateRectangleImage(for: Image(backgroundImageName),
+                                          aspectRatio: aspectRatio)
+        } else {
+            crateRectangleImage(for: Image(Assets.Image.playhouseImage),
+                                          aspectRatio: aspectRatio)
+        }
+    }
+    
+    private var foregroundGroup: some View {
+        GeometryReader { geometry in
+            VStack(spacing: Padding.stackGap){
```

Comment:
Can we switch this to .cellGap? 

Image example of the difference below.

| Before | After |
| - | - | 
|![Swift Leeds](https://user-images.githubusercontent.com/13989549/183059700-d20c4323-d8b3-495a-b695-b1f43bb22240.png)|![wift Leeds](https://user-images.githubusercontent.com/13989549/183059718-bf0e50df-3be3-43de-b7be-b281327ab3ba.png)|